lua/gitlab/reviewer/diffview.lua

```diff
@@ -126,106 +126,117 @@ M.get_location = function(range)
     return
   end
 
-  local bufnr = vim.api.nvim_get_current_buf()
-
   -- If there's a range, use the start of the visual selection, not the current line
   local current_line = range and range.start_line or vim.api.nvim_win_get_cursor(0)[1]
 
-  -- check if we are in the diffview tab
+  -- Check if we are in the diffview tab
   local tabnr = vim.api.nvim_get_current_tabpage()
   if tabnr ~= M.tabnr then
     u.notify("Line location can only be determined within reviewer window", vim.log.levels.ERROR)
     return
   end
 
-  -- check if we are in the diffview buffer
+  -- Check if we are in the diffview buffer
   local view = diffview_lib.get_current_view()
   if view == nil then
     u.notify("Could not find Diffview view", vim.log.levels.ERROR)
     return
   end
+
   local layout = view.cur_layout
-  local result = {}
-  local type
-  local is_new
-
-  if
-    layout.a.file.bufnr == bufnr
-    or (M.lines_are_same(view.cur_layout) and layout.b.file.bufnr == bufnr and range == nil)
-  then
-    result.file_name = layout.a.file.path
-    result.old_line = current_line
-    type = "old"
-    is_new = false
-  elseif layout.b.file.bufnr == bufnr then
-    result.file_name = layout.b.file.path
-    result.new_line = current_line
-    type = "new"
-    is_new = true
-  else
-    u.notify("Line location can only be determined within reviewer window")
+
+  ---@type ReviewerInfo
+  local reviewer_info = {
+    file_name = layout.a.file.path,
+    new_line = nil,
+    old_line = nil,
+    range_info = nil,
+  }
+
+  local a_win = u.get_window_id_by_buffer_id(layout.a.file.bufnr)
+  local b_win = u.get_window_id_by_buffer_id(layout.b.file.bufnr)
+  local current_win = vim.fn.win_getid()
+  local is_current_sha = current_win == b_win
+
+  if a_win == nil or b_win == nil then
+    u.notify("Error retrieving window IDs for current files", vim.log.levels.ERROR)
     return
   end
 
-  local hunks = u.parse_hunk_headers(result.file_name, state.INFO.target_branch)
-  if hunks == nil then
-    u.notify("Could not parse hunks", vim.log.levels.ERROR)
+  local current_file = M.get_current_file()
+  if current_file == nil then
+    u.notify("Error retrieving current file from Diffview", vim.log.levels.ERROR)
     return
   end
 
-  local current_line_info
-  if is_new then
-    current_line_info = u.get_lines_from_hunks(hunks, result.new_line, is_new)
-  else
-    current_line_info = u.get_lines_from_hunks(hunks, result.old_line, is_new)
+  local a_linenr = vim.api.nvim_win_get_cursor(a_win)[1]
+  local b_linenr = vim.api.nvim_win_get_cursor(b_win)[1]
```

Comment:
Here we take advantage of scroll-lock to get the line from the current and old buffers. We can use the old line even if we're in the new buffer this way.